scripts/simplify_network.py

```diff
@@ -401,12 +402,29 @@ def consense(x):
 
     busmaps = [trafo_map, simplify_links_map, stub_map]
 
-    if snakemake.config.get('clustering', {}).get('simplify', {}).get('to_substations', False):
+    cluster_config = snakemake.config.get('clustering', {}).get('simplify_network', {})
+    if cluster_config.get('to_substations', False):
         n, substation_map = aggregate_to_substations(n)
         busmaps.append(substation_map)
 
+    # treatment of outliers (nodes without a profile for considered carrier) for "cluster_network"
+    if snakemake.config.get("clustering", {}).get("cluster_network", {}).get("algorithm", "hac") == "hac":
+        carriers = cluster_config.get("feature", "solar+onwind-time").split('-')[0].split('+')
+        buses_i = list(set(n.buses.index)-set(n.generators.query("carrier in @carriers").bus))
+        n, busmap_hac = aggregate_to_substations(n, buses_i)
+        busmaps.append(busmap_hac)
+
     if snakemake.wildcards.simpl:
-        n, cluster_map = cluster(n, int(snakemake.wildcards.simpl), snakemake.config)
+        # treatment of outliers (nodes without a profile for a considered carrier) for "simplify"
+        if cluster_config.get("algorithm", "hac") == "hac":
+            carriers = cluster_config.get("feature", "solar+onwind-time").split('-')[0].split('+')
+            buses_i = list(set(n.buses.index)-set(n.generators.query("carrier in @carriers").bus))
+            n, busmap_hac = aggregate_to_substations(n, buses_i)
+            busmaps.append(busmap_hac)
```

Comment:
This code block is very similar to the one above. Could you mend it into a single function that is called at both locations?